### ISSUE TITLE
Can't load JNI library and Python bindings in same process

### ISSUE THREAD

**Issue by aadamson** [tensorflow/tensorflow#24704]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  Linux Ubuntu 16.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): Source
- TensorFlow version: 1.11.0
- Python version: 3.6.8
- Installed using virtualenv? pip? conda?: pip
- Bazel version (if compiling from source): 0.20.0
- GCC/Compiler version (if compiling from source): 5.4.0
- CUDA/cuDNN version: N/A
- GPU model and memory: N/A

**Describe the problem**

I'm trying to load both the JNI library (libtensorflow_jni.so) and the Python bindings into the same process. I've built both from source via the //tensorflow/tools/pip_package:build_pip_package and //tensorflow/java:libtensorflow_jni targets. When I load the JNI library after importing tensorflow in a Python process, I get the following error:
```python
>>> import tensorflow as tf
>>> import ctypes
>>> ctypes.CDLL('tensorflow/java/libtensorflow_jni.so')
2019-01-04 16:07:57.226191: F tensorflow/core/lib/monitoring/collection_registry.cc:77] Cannot register 2 metrics with the same name: /tensorflow/cc/saved_model/load_attempt_count
Aborted
```

**Provide the exact sequence of commands / steps that you executed before running into the problem**
- Built TensorFlow JNI library and pip package from source
- Installed whl file via pip
- Ran the steps above

My guess is one or both of _pywrap_tensorflow.so and libtensorflow_jni.so are statically linking that initialization code. Are they intended to be usable in the same process? If not, is there a way to build both objects to have them both dynamically link against the offending section?

**Comment by asimshankar** [comment]
The two libraries aren't setup to be linked together - not that we specifically don't want them to, just wasn't something we were aiming for.

I'd be curious to understand your use case here - why do you need to load both libraries? That said, the bottleneck here is that the operations and kernel registrations (the `REGISTER_OP` and `REGISTER_KERNEL` calls in source) are part of both libraries. You could perhaps try building one of them without this (e.g., removing the dependency on `//tensorflow/core:all_kernels` and `//tensorflow/core:ops` from https://github.com/tensorflow/tensorflow/blob/2703d8e84aced78a05ec7ad1ea0e5f4d0fe70f9f/tensorflow/java/src/main/native/BUILD#L43)

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 29 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by ymodak** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!

**Comment by saudet** [comment]
BTW, this is now possible with the Python-enabled builds of the JavaCPP Presets for TensorFlow.
There is an example of that there:
https://github.com/bytedeco/javacpp-presets/blob/master/tensorflow/samples/pom.xml
https://github.com/bytedeco/javacpp-presets/blob/master/tensorflow/samples/KerasMNIST.java
